packages/wallets/nufi/src/adapter.ts

```diff
@@ -0,0 +1,232 @@
+
+    private _connecting = false;
+    private _wallet: NufiWallet | null = null;
+    private _publicKey: PublicKey | null = null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' || typeof document === 'undefined'
+            ? WalletReadyState.Unsupported
+            : WalletReadyState.NotDetected;
+
+    constructor(config: NufiWalletAdapterConfig = {}) {
+        super();
+
+        if (this._readyState !== WalletReadyState.Unsupported) {
+            scopePollingDetectionStrategy(() => {
+                if (window.nufiSolana?.isNufi) {
+                    this._readyState = WalletReadyState.Installed;
+                    this.emit('readyStateChange', this._readyState);
+                    return true;
+                }
+                return false;
+            });
+        }
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get connected(): boolean {
+        return !!this._wallet?.isConnected;
+    }
+
+    get readyState(): WalletReadyState {
+        return this._readyState;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Installed) throw new WalletNotReadyError();
+
+            this._connecting = true;
+
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const wallet = window!.nufiSolana!;
+
+            if (!wallet.isConnected) {
+                try {
+                    await new Promise<void>((resolve, reject) => {
+                        const connect = () => {
+                            wallet.off('connect', connect);
+                            resolve();
+                        };
+
+                        wallet.on('connect', connect);
+
+                        wallet.connect().catch((reason: any) => {
+                            wallet.off('connect', connect);
+                            reject(reason);
+                        });
+                    });
+                } catch (error: any) {
+                    if (error instanceof WalletError) throw error;
+                    throw new WalletConnectionError(error?.message, error);
+                }
+            }
```

Comment:
It looks like this was all copied from the Phantom adapter, which has hacks to access events because of bad connection behavior. Please don't copy these things!

Reply:
Thanks, fixed.